src/fu-engine.c

```diff
@@ -1135,6 +1135,64 @@ fu_engine_modify_bios_settings(FuEngine *self,
 	return TRUE;
 }
 
+gboolean
+fu_engine_update_bios_pending_settings(FuEngine *self,
+				       GHashTable *settings,
+				       gboolean force_ro,
+				       GError **error)
+{
+	gboolean changed = FALSE;
+	GHashTableIter iter;
+	gpointer key, value;
+
+	g_return_val_if_fail(FU_IS_ENGINE(self), FALSE);
+	g_return_val_if_fail(settings != NULL, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+	g_hash_table_iter_init(&iter, settings);
+	while (g_hash_table_iter_next(&iter, &key, &value)) {
+		changed = fu_context_get_bios_setting_pending_value(self->ctx, key, value, error);
+	}
+
+	if (!changed)
+		return FALSE;
+
+	g_object_notify(G_OBJECT(self->ctx), "bios-set");
+	return TRUE;
+}
+
+static void
+fu_engine_bios_set_notify_cb(FuContext *ctx, GParamSpec *pspec, FuEngine *self)
+{
+	g_autoptr(FuBiosSettings) bios_settings = fu_context_get_bios_settings(self->ctx);
+	g_autoptr(GPtrArray) pending_bios_settings =
+	    fu_bios_settings_get_pending_values(bios_settings);
+	g_autoptr(GHashTable) bios_settings_hash =
+	    g_hash_table_new_full(g_str_hash, g_str_equal, g_free, g_free);
+	GError *local_error = NULL;
+
+	if (pending_bios_settings->len == 0)
+		return;
+
+	for (guint i = 0; i < pending_bios_settings->len; i++) {
+		FwupdBiosSetting *attr = g_ptr_array_index(pending_bios_settings, i);
+		const gchar *id = NULL;
+		const gchar *pending_value = NULL;
+
+		id = fwupd_bios_setting_get_id(attr);
+		pending_value = fwupd_bios_setting_get_pending_value(attr);
+		if (pending_value == NULL)
+			continue;
+
+		g_info("BIOS settings: %s is updating to %s", id, pending_value);
```

Comment:
this seems needlessly noisy I'd think, do you really still need it?

Reply:
Okay. I'll remove it.